src/pages/RoomInvitePage.tsx

```diff
@@ -26,77 +27,60 @@ import * as Report from '@userActions/Report';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {PersonalDetailsList, Policy} from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import type {WithReportOrNotFoundProps} from './home/report/withReportOrNotFound';
 import withReportOrNotFound from './home/report/withReportOrNotFound';
-import personalDetailsPropType from './personalDetailsPropType';
-import reportPropTypes from './reportPropTypes';
-import {policyDefaultProps, policyPropTypes} from './workspace/withPolicy';
-
-const propTypes = {
-    /** Beta features list */
-    betas: PropTypes.arrayOf(PropTypes.string),
 
+type RoomInvitePageOnyxProps = {
     /** All of the personal details for everyone */
-    personalDetails: PropTypes.objectOf(personalDetailsPropType),
-
-    /** URL Route params */
-    route: PropTypes.shape({
-        /** Params from the URL path */
-        params: PropTypes.shape({
-            /** policyID passed via route: /workspace/:policyID/invite */
-            policyID: PropTypes.string,
-        }),
-    }).isRequired,
-
-    /** The report currently being looked at */
-    report: reportPropTypes.isRequired,
-
-    /** The policies which the user has access to and which the report could be tied to */
-    policies: PropTypes.shape({
-        /** ID of the policy */
-        id: PropTypes.string,
-    }).isRequired,
-
-    ...policyPropTypes,
+    personalDetails: OnyxEntry<PersonalDetailsList>;
 };
 
-const defaultProps = {
-    personalDetails: {},
-    betas: [],
-    ...policyDefaultProps,
-};
+type RoomInvitePageProps = RoomInvitePageOnyxProps & WithReportOrNotFoundProps;
 
-function RoomInvitePage(props) {
+type Sections = Array<SectionListData<OptionsListUtils.MemberForList, Section<OptionsListUtils.MemberForList>>>;
+
+function RoomInvitePage({betas, personalDetails, report, policies}: RoomInvitePageProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const [searchTerm, setSearchTerm] = useState('');
-    const [selectedOptions, setSelectedOptions] = useState([]);
-    const [personalDetails, setPersonalDetails] = useState([]);
-    const [userToInvite, setUserToInvite] = useState(null);
+    const [selectedOptions, setSelectedOptions] = useState<ReportUtils.OptionData[]>([]);
+    const [invitePersonalDetails, setPersonalDetails] = useState<ReportUtils.OptionData[]>([]);
```

Comment:
should we rename the set state method? I was a bit confused when tracking where was `invitePersonalDetails` updated.

```suggestion
    const [invitePersonalDetails, setInvitePersonalDetails] = useState<ReportUtils.OptionData[]>([]);
```

Reply:
Good point, should have renamed it when I was renaming personalDetails into invitePersonalDetails